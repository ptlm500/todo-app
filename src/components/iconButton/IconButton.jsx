import styled from 'styled-components';
import PropTypes from 'prop-types';
import { theme } from '../../styles';

const LARGE = 'large';
const SMALL = 'small';
const LARGE_SIZE = '3rem';
const SMALL_SIZE = '1rem';
const sizes = {
  'large': LARGE_SIZE,
  'small': SMALL_SIZE
};

function getSize(size) {
  return sizes[size];
}

const IconButton = styled.button`
  background: none;
  border: none;
  width: ${props => getSize(props.size)};
  height: ${props => getSize(props.size)};
  padding: 0;
  cursor: pointer;
  transition: ${theme.transition.long};

  > svg {
    width: 100%;
    height: 100%;
  }

  &:hover {
    transform: scale(1.2) rotate(1turn);
  }
`;

IconButton.propTypes = {
  size: PropTypes.oneOf([LARGE, SMALL])
};

IconButton.defaultProps = {
  size: LARGE
};

export default IconButton;
