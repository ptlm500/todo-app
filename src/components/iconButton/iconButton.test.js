import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from './IconButton';
import 'jest-styled-components';

describe('IconButton', () => {
  it('renders an IconButton', () => {
    const tree = renderer.create(<IconButton />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a small IconButton', () => {
    const tree = renderer.create(<IconButton size='small' />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a large IconButton', () => {
    const tree = renderer.create(<IconButton size='large' />);
    expect(tree).toMatchSnapshot();
  });
});
