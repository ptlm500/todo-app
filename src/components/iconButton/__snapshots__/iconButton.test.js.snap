// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders a large IconButton 1`] = `
.c0 {
  background: none;
  border: none;
  width: 3rem;
  height: 3rem;
  padding: 0;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 > svg {
  width: 100%;
  height: 100%;
}

.c0:hover {
  -webkit-transform: scale(1.2) rotate(1turn);
  -ms-transform: scale(1.2) rotate(1turn);
  transform: scale(1.2) rotate(1turn);
}

<button
  className="c0"
  size="large"
/>
`;

exports[`IconButton renders a small IconButton 1`] = `
.c0 {
  background: none;
  border: none;
  width: 1rem;
  height: 1rem;
  padding: 0;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 > svg {
  width: 100%;
  height: 100%;
}

.c0:hover {
  -webkit-transform: scale(1.2) rotate(1turn);
  -ms-transform: scale(1.2) rotate(1turn);
  transform: scale(1.2) rotate(1turn);
}

<button
  className="c0"
  size="small"
/>
`;

exports[`IconButton renders an IconButton 1`] = `
.c0 {
  background: none;
  border: none;
  width: 3rem;
  height: 3rem;
  padding: 0;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 > svg {
  width: 100%;
  height: 100%;
}

.c0:hover {
  -webkit-transform: scale(1.2) rotate(1turn);
  -ms-transform: scale(1.2) rotate(1turn);
  transform: scale(1.2) rotate(1turn);
}

<button
  className="c0"
  size="large"
/>
`;
